src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -3837,6 +3837,213 @@ static void Main()
             Assert.True(type.IsNativeIntegerType);
         }
 
+        [Theory, CombinatorialData]
+        [WorkItem(60337, "https://github.com/dotnet/roslyn/issues/60337")]
+        public void NameOfNintSpeculation_01([CombinatorialValues("always", "never")] string runNullableAnalysis)
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        var x = nameof(nint);
+    }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9.WithFeature("run-nullable-analysis", runNullableAnalysis));
+            comp.VerifyDiagnostics(
+                // (6,24): error CS0103: The name 'nint' does not exist in the current context
+                //         var x = nameof(nint);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "nint").WithArguments("nint").WithLocation(6, 24)
+            );
+            var tree = comp.SyntaxTrees.Single();
+            var root = tree.GetRoot();
+            var existingNint = root.DescendantNodes().OfType<ArgumentSyntax>().Single().Expression;
+            var existingNameOf = root.DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+            var model = comp.GetSemanticModel(tree);
+            var nameofToSpeculate = (InvocationExpressionSyntax)SyntaxFactory.ParseExpression("nameof(nint)");
+
+            var speculativeSymbolInfo = model.GetSpeculativeSymbolInfo(existingNameOf.SpanStart, nameofToSpeculate.ArgumentList.Arguments[0].Expression, SpeculativeBindingOption.BindAsExpression);
```

Comment:
>nameofToSpeculate.ArgumentList.Arguments[0].Expression [](http://example.com/codeflow?start=97&length=54)

It is really not obvious what we are speculating on. In cases like this, we at least assert textual representation of the node in order to confirm that we are testing what we intend to test and to make the intent obvious for reviewers and future readers.